### ISSUE TITLE
Any way to get evaluation results during training?

### ISSUE THREAD

**Issue by 2ec** [tloen/alpaca-lora#268]
It seems that during training the Trainer only prints and logs the training loss.

To get an idea of how the model trains during fine-tuning, it would be useful to also print and log evaluations loss for each training step to better catch overtraining.

Does someone have a good way to log evaluation loss alongside training loss?

**Comment by lywinged** [comment]
Here is an example if you want to check the train and eval loss for each logging steps.

args=transformers.TrainingArguments(
logging_steps=2,
eval_steps=2
)

Then it went like this:
{'loss': 2.3911, 'learning_rate': 0.00029922480620155035, 'epoch': 0.01}    
{'eval_loss': 2.2206013202667236, 'eval_runtime': 274.3799, 'eval_samples_pe
r_second': 7.289, 'eval_steps_per_second': 0.911, 'epoch': 0.01}            
{'loss': 2.2272, 'learning_rate': 0.00029767441860465116, 'epoch': 0.01}    
{'eval_loss': 2.098778247833252, 'eval_runtime': 276.9351, 'eval_samples_per_second': 7.222, 'eval_steps_per_second': 0.903, 'epoch': 0.01}             
  1%|▎                                  | 4/388

**Comment by 2ec** [comment]
@lywinged Thanks a lot! 🥳

**Comment by rchennnn** [comment]
also need to set evaluation_strategy to be one of "steps" or "epochs". See tranformers.training_args.py
